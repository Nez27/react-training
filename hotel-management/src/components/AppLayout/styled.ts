import styled from 'styled-components';

const StyledAppLayout = styled.div`
  display: grid;
  grid-template-columns: 300px 1fr;
  grid-template-rows: auto 1fr;
  
  height: 100vh;
`;

const Main = styled.main`
  border: 1px solid var(--border-color);

  overflow: scroll;
`;

export { StyledAppLayout, Main };
