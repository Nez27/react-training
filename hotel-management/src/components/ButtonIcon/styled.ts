import styled, { css } from 'styled-components';

interface IStyledButtonIcon {
  isHaveChildren: boolean;
  style?: {
    fontSize?: string;
    backgroundColor?: string;
    color?: string;
  };
  iconStyle?: {
    color?: string;
    size?: string;
  };
}

const StyledButtonIcon = styled.button<IStyledButtonIcon>`
  background: none;

  padding: 10px;
  transition: all 0.2s;

  border: none;

  ${(props) =>
    props.style?.fontSize &&
    css`
      font-size: ${props.style.fontSize};
    `}

  ${(props) =>
    props.style?.fontSize &&
    css`
      font-size: ${props.style.fontSize};
    `}

  ${(props) =>
    props.color &&
    css`
      font-size: ${props.color};
    `}

  ${(props) =>
    props.isHaveChildren
      ? css`
          display: flex;
          align-items: center;
          gap: 10px;

          text-align: left;
          width: 100%;
        `
      : css`
          border-radius: var(--radius-md);
        `}

  &:hover {
    background-color: var(--hover-background-color);
  }

  & svg {
    ${(props) =>
      props.iconStyle?.size &&
      css`
        width: ${props.iconStyle.size};
        height: ${props.iconStyle.size};
      `}

    ${(props) =>
      props.iconStyle?.color &&
      css`
        color: ${props.iconStyle.color};
      `}
    transition: all 0.3s;
  }
`;

StyledButtonIcon.defaultProps = {
  style: {
    fontSize: '14px',
    backgroundColor: 'var(--primary-color)',
    color: '#000',
  },
  iconStyle: {
    color: '#000',
    size: '14px',
  },
};

export { StyledButtonIcon };
