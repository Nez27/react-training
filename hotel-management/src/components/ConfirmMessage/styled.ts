import styled from "styled-components";

const StyledConfirmDelete = styled.div`
  width: 450px;

  & p {
    font-size: var(--fs-sm);
    text-align: center;
    margin-bottom: 30px;
  }

  & div {
    display: flex;
    justify-content: space-around;

    padding-inline: 100px;
  }
`;

export { StyledConfirmDelete };
