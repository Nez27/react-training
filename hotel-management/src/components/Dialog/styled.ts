import styled from 'styled-components';

const StyledDialog = styled.dialog`
  border-radius: var(--radius-sm);
  border-color: var(--border-color);
`;

const StyledTitle = styled.p`
  font-size: var(--fs-md);
  text-transform: capitalize;
  font-weight: 600;

  padding: 20px 40px;
  border-bottom: 1px solid var(--border-color);

  text-align: center;
`;

const StyledBody = styled.div`
  padding: 10px 20px;
`;

export { StyledDialog, StyledTitle, StyledBody };
