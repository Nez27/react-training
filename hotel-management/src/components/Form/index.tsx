import { FormEvent, ReactNode } from 'react';

// Styled
import { StyledActionBtn, StyledForm } from './styled';

interface IFormProps {
  children: ReactNode;
  onSubmit: (event: FormEvent<HTMLFormElement>) => void;
  id?: string;
}

const Form = ({ children, onSubmit, id }: IFormProps) => {
  return (
    <StyledForm onSubmit={onSubmit} id={id}>
      {children}
    </StyledForm>
  );
};

Form.Action = StyledActionBtn;

export default Form;
