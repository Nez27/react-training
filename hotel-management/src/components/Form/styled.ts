import Button from '@commonStyle/Button';
import styled from 'styled-components';

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  gap: 15px;
`;

const StyledActionBtn = styled.div`
  display: flex;
  justify-content: space-around;
  flex-direction: row;
  gap: 100px;
`;

const StyledFormRow = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 100px;
`;

const Label = styled.label`
  font-size: var(--fs-sm);
  text-transform: capitalize;
`;

const Error = styled.p`
  color: var(--error-text);
  font-size: var(--fs-sm-2x);

  margin-top: 5px;
  padding-inline: 5px;

  width: 220px;
`;

const FormBtn = styled(Button)`
  width: 100%;

  &:disabled,
  &[disabled] {
    background-color: var(--disabled-btn-color);
  }
`;

export { StyledForm, StyledActionBtn, StyledFormRow, Label, Error, FormBtn };
