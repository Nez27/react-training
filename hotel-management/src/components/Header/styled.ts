import styled from 'styled-components';

const StyledHeader = styled.header`
  border-bottom: 1px solid var(--border-color);
  padding: 20px 40px;

  display: flex;
  justify-content: flex-end;
  align-items: center;
  gap: 30px;
`;

export { StyledHeader };
