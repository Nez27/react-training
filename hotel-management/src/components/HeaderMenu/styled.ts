import styled from 'styled-components';

const StyledHeaderMenu = styled.ul`
  display: flex;
  gap: 10px;
`;

export { StyledHeaderMenu };
