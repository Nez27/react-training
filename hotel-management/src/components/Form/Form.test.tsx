import renderer from 'react-test-renderer';

// Components
import Form from '.';
import Input from '@commonStyle/Input';

// Styled
import { FormBtn } from '@page/User/styled';

describe('Form testing snapshot', () => {
  const isDisable = true;
  const errorMessage = 'This is error message';
  const handleOnSubmit = () => {
    console.log('Confirm');
  };
  const handleOnClick = () => {
    console.log('On Click');
  };

  const wrapper = renderer.create(
    <Form onSubmit={handleOnSubmit}>
      <Form.Row label="Name" error={errorMessage}>
        <Input type="text" id="name" />
      </Form.Row>

      <Form.Row label="Price" error={errorMessage}>
        <Input type="text" id="price" />
      </Form.Row>

      <Form.Action>
        <FormBtn type="submit" name="submit" disabled={isDisable}>
          Confirm
        </FormBtn>
        <FormBtn type="button" variations="secondary" onClick={handleOnClick}>
          Close
        </FormBtn>
      </Form.Action>
    </Form>
  );

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
