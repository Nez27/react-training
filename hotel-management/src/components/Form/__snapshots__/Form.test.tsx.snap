// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form  Should render correctly 1`] = `
<form
  className="sc-bdnyFh bzxSLR"
  onSubmit={[MockFunction]}
>
  <div
    className="sc-dlniIP gZHESL"
  >
    <label
      className="sc-hKFymg ddFGOI"
    >
      Name
    </label>
    <div>
      <input
        className="sc-jSFipO enjwJE"
        id="name"
        type="text"
      />
      <p
        className="sc-eCAqax bvOWPP"
      >
        This is error message
      </p>
    </div>
  </div>
  <div
    className="sc-dlniIP gZHESL"
  >
    <label
      className="sc-hKFymg ddFGOI"
    >
      Price
    </label>
    <div>
      <input
        className="sc-jSFipO enjwJE"
        id="price"
        type="text"
      />
      <p
        className="sc-eCAqax bvOWPP"
      >
        This is error message
      </p>
    </div>
  </div>
  <div
    className="sc-gtsqUy idhRSV"
  >
    <button
      className="sc-gKAaef sc-jrsKJM gifCOZ KLoQf"
      disabled={true}
      name="submit"
      type="submit"
      variations="primary"
    >
      Confirm
    </button>
    <button
      className="sc-gKAaef sc-jrsKJM gbgZUe KLoQf"
      onClick={[MockFunction]}
      type="button"
      variations="secondary"
    >
      Close
    </button>
  </div>
</form>
`;
